knxdclient/encoding.py

```diff
@@ -104,6 +106,7 @@ def from_datetime(cls, value: datetime.datetime):
     KNXDPT.DATE_TIME: (datetime.datetime, datetime.date, datetime.time),
     KNXDPT.ENUM8: (int, enum.Enum),
     KNXDPT.VARSTRING: str,
+    KNXDPT.COLOUR_RGB: bytes,
```

Comment:
In my opinion, it is somehow strange to use the `bytes` type for encoding a 24bit RGB value. I would prefer to use a tuple `(int, int, int)` instead.

Reply:
Would a tuple of `(bytes, bytes, bytes)` not be better than ints, given each value is from 0-255?

Reply:
No, not really. `bytes` is – as the name already suggests – an array-like structure of multiple bytes. It can be accessed like list in Python, each entry being an `int` in the range 0-255. Unfortunately, Python does not have built-in fixed-size integer types like `uint8_t` in C/C++ etc. So the next best we can do is working with `(int, int, int)` and manually write code to check that the integers are in the expected range.

Reply:
I keep hitting a brick wall if I try to define this DPT as `KNXDPT.COLOUR_RGB: (int, int, int),`

Each time the bus sends a telegram as 232.600 it's discarded with: 

`Expected a (<class 'int'>, <class 'int'>, <class 'int'>) for KNX COLOUR_RGB, not b'\xdd\xee\xff'`.

(In this example I'd sent #DDEEFF from ETS).

It's much happier if I go back to `KNXDPT.COLOUR_RGB: bytes,` and then in `decode_value` I return a tuple:
`return (val[0] , val[1] , val[2])`.

Am I missing something obvious? (I initially chose to define it as `bytes` because with only a couple of exceptions that's how you were handling all other DPTs, so I simply followed suit).

Once we can sort out the structure I'll add the data validation for encoding and then get the tests to align.

On that former point, are you OK with `encode_value` encoding an input as 0 if input is <0 and 255 if it's > 255, or throw a ValueError instead?

Reply:
> Am I missing something obvious? (I initially chose to define it as bytes because with only a couple of exceptions that's how you were handling all other DPTs, so I simply followed suit).

Maybe we're talking about two different things: The `DPT_ENCODING` (e.g. the data type of the *encoded* value for the KNX bus) of the DPT 232 *must* be `bytes`. This type defines how the KNX APDU packet (Application Data Unit) is structured, and there are (only) two possibilities according to the KNX standard, which we represent as `bytes` or `int` object respectively.

On the other hand, I suggested to use `(int, int, int)` as the `DPT_PYTHON_REPRESENTATION` for DPT 232, which is the representation of the *decoded* value on the "Python side". We can chose this type freely, but need to adjust `decode_value()` and `encode_value()` such that they convert between the KNX bytes representation and our Python representation accordingly.

My suggestion for DPT 232 was:
- `encode_value()`: `(int, int, int) -> bytes`
- `decode_value()`: `bytes -> (int, int, int)`

Since a `bytes` object of length 3 should behave very similar to an `(int, int, int)` tuple, the implementation should be straight forward, by simply using the "type cast" constructors of the two types:

In `encode_value()`:
```python
return bytes(val)
```

In `decode_value()`:
```python
return tuple(val)
```

Of course we could also add some additional error handling: checking that the integers are in range 0..255 in `encode_value()` and checking that the incoming `bytes` object actually has length 3 in `decode_value()`. The former check is already done by the `bytes()` constructor, so I'd only add the latter one.

Reply:
Thanks for clarifying that. I'm making some progress: I can decode OK, but still can't encode.

I decided to add a 'type' to the error message at the top of `encode_value`, so it tells me what I'm doing wrong, but with the explanation I'm none the wiser:

```
raise TypeError(f"Cannot use {repr(value)} as a KNX {t.name}, since it is a {type(value)} not a {DPT_PYTHON_REPRESENTATION[t]}")
```

Run the tests:
```
def test_dpt232_conversion(self) -> None:
        self.assertEqual(bytes([0x7F, 0x00, 0x20]), knxdclient.encode_value((0x7F, 0x00, 0x20), knxdclient.KNXDPT.COLOUR_RGB))
```

OK, so how on earth is the value to be encoded turning into a *string*?

```
TypeError: Cannot use (0x7F, 0x00, 0x20) as a KNX COLOUR_RGB, since it is a <class 'str'> not a (<class 'int'>, <class 'int'>, <class 'int'>)
```

I've tried all sorts of ways to cajole it without any success, even removing the spaces after the comma separating each member of the tuple.

I even got desperate and tried to cast each member, but it's STILL a string!
```
self.assertEqual(bytes([0x7F, 0x00, 0x20]), knxdclient.encode_value((int(17),int(254),int(99)), knxdclient.KNXDPT.COLOUR_RGB))
```